File: back/src/main/java/csh/back/domain/trip/post/dto/response/PostTimeLineResponse.java
package csh.back.domain.trip.post.dto.response;

import csh.back.domain.trip.post.entity.Post;

import java.time.LocalDateTime;

public record PostTimeLineResponse(
        LocalDateTime startTime,
        LocalDateTime endTime,
        Long timeLineId,      // ← 추가: 일정 슬롯이면 값, 빈 칸이면 null
        String confirmedPlaceName,
        boolean isTaken

) {
    public static PostTimeLineResponse of(
            LocalDateTime startTime,
            LocalDateTime endTime,
            Long timeLineId,      // ← 추가: 일정 슬롯이면 값, 빈 칸이면 null
            String confirmedPlaceName,
            boolean isTaken) {
        return new PostTimeLineResponse(startTime, endTime, timeLineId, confirmedPlaceName, isTaken);
    }
}

File: back/src/main/kotlin/csh/back/domain/trip/post/dto/response/PostTimelineResponse.kt
package csh.back.domain.trip.post.dto.response

import java.time.LocalDateTime

data class PostTimelineResponse(
    val startTime: LocalDateTime,
    val endTime: LocalDateTime,
    val timelineId: Long?,
    val confirmedPlaceName: String?,
    val isTaken: Boolean
) {
    companion object {
        @JvmStatic
        fun of(
            startTime: LocalDateTime,
            endTime: LocalDateTime,
            timelineId: Long?,
            confirmedPlaceName: String?,
            isTaken: Boolean
        ) = PostTimelineResponse(startTime, endTime, timelineId, confirmedPlaceName, isTaken)
    }
}
